test/cert-ceremonies/generate.go

```diff
@@ -73,6 +73,12 @@ func genCert(path string) error {
 }
 
 func main() {
+	// If one of the output files already exists, assume this ran once
+	// already for the container and don't re-run.
+	if _, err := os.Stat("/tmp/root-signing-pub-rsa.pem"); err == nil {
```

Comment:
Small nit. There are circumstances where `err != nil` but a failure may still be encountered during execution. We could mitigate this or make it clearer to the operator with some checks. 

Additionally, though highly unlikely, `os.Stat` works on files and directories all the same. In the event of operator error or configuration management gone awry, a directory at `/tmp/root-signing-pub-rsa.pem` would lead to this operation being skipped erroneously.

I believe the code below would be able to mitigate both of these and give the operator some additional information in the event of failure.

```go
func main() {
	loc, err := os.Stat("/tmp/root-signing-pub-rsa.pem")
	if err != nil {
		cmd.FailOnError(err, "while checking if...")
	} else {
		if loc.IsDir() {
			cmd.FailOnError(fmt.Errorf("%s: is a directory should be a file", loc.Name()), "while checking...")
		}
		fmt.Println("skipping certificate generation: already exists")
	}
}
```

Reply:
Additionally, in the event that `os.Stat` returns an `error` we can also check whether the underlying error is due to a missing file:

```go
func main() {
	loc, err := os.Stat("/tmp/root-signing-pub-rsa.pem")
	if err != nil {
		if !os.IsNotExist(err) {
			// the error encountered did not pertain the file itself being absent
			cmd.FailOnError(err, "while checking if an output file is already present")
		}
	} else {
		if loc.IsDir() {
			// the path provided is a directory, not a file
			cmd.FailOnError(fmt.Errorf("%s: is a directory should be a file", loc.Name()),
				"while checking if an output file is already present")
		}
		// err == nil and loc is not a directory
		fmt.Println("skipping certificate generation: already exists")
		return
	}
}
```

Reply:
I think I was failing to see the forest for the trees on this one.

Reply:
I think the point about files vs directories is a good one. Maybe the simplest way to do it is
```golang
loc, err := os.Stat("path")
if err == nil && loc.IsRegular() {
    // proceed as normal
}
```